clients/src/main/java/org/apache/kafka/common/network/NetworkSend.java

```diff
@@ -23,20 +23,19 @@
  */
 public class NetworkSend extends ByteBufferSend {
 
-    public NetworkSend(String destination, ByteBuffer... buffers) {
-        super(destination, sizeDelimit(buffers));
+    public NetworkSend(String destination, ByteBuffer buffer) {
```

Comment:
I see that we don't use the `...` version anywhere for `NetworkSend`, but the drawback to this change is that constructors become inconsistent for the different classes. Not a big deal, but might be worth considering. (Admittedly, the updated implementation is also probably a bit more efficient since it avoids a loop & dynamically sized arrays.)